doc/tutorial.md

```diff
@@ -799,3 +800,203 @@ Statistics are supported for debug purpose. Note `recvfrom(2)` is kept untouched
 It's useful to send the data back by socket. Pls note UDP socket is connect-less, one `socket-fd` can be used to communicate with different peers.
 
 Actually, we use private IP option to implement `UOA`, pls check the details in [uoa.md](../uoa/uoa.md).
+
+<a id='Ubuntu16.04'/>
+
+# DPVS in Ubuntu16.04
+
+
+### DPDK build and install
+
+Before DPDK build and install ,Fix code for ubuntu in vm
+
+```
+$ cd dpdk-stable-17.05.2/
+$ sed -i "s/pci_intx_mask_supported(dev)/pci_intx_mask_supported(dev)||true/g" lib/librte_eal/linuxapp/igb_uio/igb_uio.c
+```
+
+Now to set up DPDK hugepage, our test environment is NUMA system. For single-node system pls refer the [link](http://dpdk.org/doc/guides/linux_gsg/sys_reqs.html).
+
+```bash
+$ # for single node machine
+$ echo 1024 > /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages
+```
+
+## Build DPVS
+
+
+
+> may need install dependencies, like `openssl`, `popt` and `numactl`, e.g., ` apt-get install libpopt-dev libssl-dev libnuma-dev` (Ubuntu).
+
+
+
+## Launch DPVS
+Now, `dpvs.conf` must be put at `/etc/dpvs.conf`, just copy it from `conf/dpvs.conf.single-nic.sample`.
+
+```bash
+$ cp conf/dpvs.conf.single-nic.sample /etc/dpvs.conf
+```
+do some changes for vm ( e1000 do not support fdir ,use single cpu )
```

Comment:
The NIC for Ubuntu may not support flow-director (fdir), for that case, pls use 'single worker'.